### PR TITLE
updates to build with qt 5.10, refactor scriptapitest

### PR DESCRIPTION
**Pull request by scottminter**
DO NOT MERGE YET!

### CODE REVIEW

**Comment by scottminter** [comment]
DO NOT MERGE

File: scriptapi/scriptapi.pro [hunk 1]
```diff
@@ -274,30 +310,28 @@ SOURCES += setupscriptapi.cpp \
...
+              qwebsecurityoriginproto.h \
+              qwebsettingsproto.h \
+              qwebviewproto.h
+  SOURCES += qwebelementcollectionproto.cpp \
+              qwebelementproto.cpp \
+              qwebframeproto.cpp \
+              qwebpageproto.cpp \
+              qwebsecurityoriginproto.cpp \
+              qwebsettingsproto.cpp \
+              qwebviewproto.cpp
```
Comment (gilmoskowitz) [inline scriptapi/scriptapi.pro]: - This looks like a duplication of the content at lines 165+ above. We only need one or the other.
- In a separate PR on the 4_11_x branch, please modify the functions in the deprecated .cpp files listed here to [report their deprecated status](https://github.com/xtuple/qt-client/blob/4_11_x/scriptapi/setupscriptapi.cpp#L320-L340) like the [script toolbox](https://github.com/xtuple/qt-client/blob/4_11_x/guiclient/scripttoolbox.cpp#L296) now does.

File: scriptapi/setupscriptapi.cpp [hunk 2]
```diff
@@ -135,28 +136,44 @@
...
+
+#if QT_VERSION <= 0x050600
+  #include "qwebelementcollectionproto.h"
+  #include "qwebelementproto.h"
+  #include "qwebframeproto.h"
+  #include "qwebpageproto.h"
+  #include "qwebsecurityoriginproto.h"
+  #include "qwebsettingsproto.h"
+  #include "qwebviewproto.h"
+#endif
```
Comment (gilmoskowitz) [inline scriptapi/setupscriptapi.cpp]: `0x050600` is always less than `0x050900` so we only need one of the checks above. Make sure it matches the similar check in the `.pro` file, whichever way you decide to go.

File: scriptapi/setupscriptapi.cpp [hunk 3]
```diff
@@ -135,28 +136,44 @@
...
+
+#if QT_VERSION > 0x050900
```
Comment (gilmoskowitz) [inline scriptapi/setupscriptapi.cpp]: This should be `>=`

**Pull request merged**